File: frontend/console/src/api/logic-object.js
import Axios from "@/http";

export function createLogicObject(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/logic-object', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveLogicObject(logicObjectId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/logic-object/${ logicObjectId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateLogicObject(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/logic-object', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteLogicObject(logicObjectId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/logic-object/${ logicObjectId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteLogicObjectWithFlag(logicObjectId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/logic-object/${ logicObjectId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllLogicObjects() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/logic-object/all').then(res => {
            resolve(res)
        })
    })
}

export function queryLogicObject (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/logic-object/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function getLogicObjectPath(logicObjectId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/path/logic-object/${ logicObjectId }`).then(res => {
            resolve(res)
        })
    })
}

export function getSignalPath(signalId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/path/signal/${ signalId }`).then(res => {
            resolve(res)
        })
    })
}

File: little-bee/controller/router/logic_object.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_LOGIC_OBJECT = "/config/logic-object"
const URL_ROUTER_LOGIC_OBJECT_BY_ID = "/config/logic-object/:logic-object-id"
const URL_ROUTER_LOGIC_OBJECT_QUERY = "/config/logic-object/query"
const URL_ROUTER_LOGIC_OBJECT_ALL = "/config/logic-object/all"

const URL_ROUTER_PATH_BY_LOGIC_OBJECT_ID = "/config/path/logic-object/:logic-object-id"
const URL_ROUTER_PATH_BY_SIGNAL_ID = "/config/path/signal/:signal-id"

func LogicObjectRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_LOGIC_OBJECT, config.CreateLogicObject)
	r.GET(URL_ROUTER_LOGIC_OBJECT_BY_ID, config.RetrieveLogicObject)
	r.PUT(URL_ROUTER_LOGIC_OBJECT, config.UpdateLogicObject)
	r.DELETE(URL_ROUTER_LOGIC_OBJECT_BY_ID, config.DeleteLogicObject)
	r.PUT(URL_ROUTER_LOGIC_OBJECT_BY_ID, config.DeleteLogicObjectWithFlag)
	r.POST(URL_ROUTER_LOGIC_OBJECT_QUERY, config.QueryLogicObject)
	r.GET(URL_ROUTER_LOGIC_OBJECT_ALL, config.GetAllLogicObjects)
	r.GET(URL_ROUTER_PATH_BY_LOGIC_OBJECT_ID,config.GetPathByLogicObjectId)
	r.GET(URL_ROUTER_PATH_BY_SIGNAL_ID,config.GetPathBySignalId)
	return r
}